packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.tsx

```diff
@@ -244,44 +240,19 @@ export class AtomicRefineModal implements InitializableComponent {
     );
   }
 
-  private dismiss() {
-    setTimeout(() => {
-      this.modalStatus = 'closed';
-    }, 500);
-    this.modalStatus = 'beingClosed';
-  }
-
   public render() {
-    if (this.modalStatus === 'closed') {
-      return;
-    }
-
-    const isOpened = this.modalStatus === 'opened';
-
     return (
-      <atomic-focus-trap active={isOpened}>
-        <div
-          part="container"
-          class={`fixed flex flex-col justify-between bg-background text-on-background left-0 top-0 right-0 bottom-0 z-10 ${
-            isOpened
-              ? 'animate-scaleUpRefineModal'
-              : 'animate-slideDownRefineModal'
-          }`}
-          aria-modal={isOpened.toString()}
-        >
-          <div class="px-6">{this.renderHeader()}</div>
-          <hr class="border-neutral"></hr>
-          <div class="overflow-auto px-6 grow">
-            <div class="adjust-for-scroll-bar">
-              <aside class="centered">
-                {this.renderSort()}
-                {this.renderFilters()}
-              </aside>
-            </div>
-          </div>
-          {this.renderFooter()}
-        </div>
-      </atomic-focus-trap>
+      <atomic-modal
+        isOpen={this.isOpen}
+        source={this.openButton}
+        close={() => (this.isOpen = false)}
+        onAnimationEnded={() => this.onAnimationEnded()}
+        exportparts="container,header,header-wrapper,header-ruler,body,body-wrapper,footer,footer-wrapper,footer-wrapper"
+      >
+        {this.renderHeader()}
+        {this.renderBody()}
```

Comment:
I extracted `renderBody` to be consistent with the other two render functions. All three return a div with `slot=`.